### PR TITLE
Feature/add settings and form

### PR DESCRIPTION
**Pull request by stroich**
## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description
Added personal settings

### CODE REVIEW

File: src/components/ModalSetting/verificationRules.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import {RegisterOptions} from 'react-hook-form';
+import {SettingData} from './ModalSetting';
+
+type rulesType = RegisterOptions<SettingData, keyof SettingData>;
```
Comment (anion155) [inline src/components/ModalSetting/verificationRules.ts]: Type names more often named with PascalCase

Comment (stroich) [reply src/components/ModalSetting/verificationRules.ts]: This item has been corrected

File: src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+};
+
+type DropdownProps = {
+  data: Array<ItemType>;
+  onChange: (value: string) => void;
+  value: string;
+};
+
+function Dropdown({data, onChange, value}: DropdownProps) {
+  const initialSelected = value ? value : 'Select options';
```
Comment (anion155) [inline src/components/Dropdown/Dropdown.tsx]:
```suggestion
  const initialSelected = value || 'Select options';
```

File: src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  const [selected, setSelected] = useState(initialSelected);
+  const [isOpenOptions, setIsOpenOptions] = useState(false);
+  const {colors} = useContext(ThemeContext);
+  const [inputValue, setInputValue] = useState('');
+  const [sortedData, setSortedData] = useState(data);
```
Comment (anion155) [inline src/components/Dropdown/Dropdown.tsx]:
```suggestion
  // const [sorting, setSorting] = useState('');
  const sortedData = useMemo(() => data.filter(el => el.value.startsWith(inputValue)) , [inputValue])
```

**Review by anion155 — APPROVED** [review]

**Pull request merged**